### DISCUSSION TITLE
Authentication for Nextra Doc Theme

### DISCUSSION THREAD

**Discussion by iamgideonai** [shuding/nextra#2446] [Q&A]
HI,

Is it possible to apply authentication to doc theme? If so, could anyone please instruct me on how to add any of the authentication services (Next Auth / AWS Cognito / Clerk / etc.,)

Thank You!!

**Comment by tswymer** [comment]
You can use Next.js middleware to add authentication to your Nextra project with Next Auth:

````typescript
/*
 * This file is responsbile for all the auth magic for the documentation.
 * It ensures that the user has a session before we show them anything on the page.
 *
 * This is all done server-side.
 *
 * Useful links:
 * @see https://nextjs.org/docs/app/building-your-application/routing/middleware
 * @see https://next-auth.js.org/configuration/nextjs#basic-usage
 */

import { withAuth } from 'next-auth/middleware';

export default withAuth({
    callbacks: {
        authorized({ token }) {
            // If we are in dev mode, don't require auth.
            if (process.env.NODE_ENV === 'development') return true;

            // If there is a session token, allow login
            if (token) return true;

            // By default, don't allow login.
            return false;
        },
    },
});

````

**Reply by Reddybrekkers** [reply]
@tswymer - thanks for this. Do I simply instlal the middleware into the theme and then insert this on to every page? I'm a non-technical person attempting to get this work (and very much need it to).

**Comment by crguezl** [comment]
@Reddybrekkers, 

You can find an example of nextra with authentication using `next-auth` at my repo <https://github.com/crguezl/pl-nextra.git>. 
Additionally you have to go to the Oauth provider (if using GitHub  to <https://github.com/settings/developers>) and setup your OAuth app. I made a [summary here](https://ull-pl.vercel.app/nextra-playground/authorization/next-auth-tutorial).

Suggestion of how to improve it are welcome!

**Comment by curlhash** [comment]
Just to clarify a doubt I have, this will also work for partial routes and not only for entire app? Example let's say I have 10 routes (or 10 pages) and I want to keep 5 of those routes public and rest 5 as private. Can anyone please clarify this?

**Comment by dimaMachina** [comment]
https://github.com/shuding/nextra/issues/1121#issuecomment-1374958546
